spec/support/dependency_helpers.rb

```diff
@@ -1,69 +1,106 @@
+  # @option :skip_install [Boolean, nil] skip running gem install, default is value of skip_build
+  # @option :version [String, nil] version for the gem, default is "1.0.0"
+  # @return void
+  def build_gem(gem_name, opts = {})
     ENV["GEM_HOME"] = TMP_GEM_ROOT
+    version = "1.0.0" # default
 
-    unless File.exist? "#{TMP_GEM_ROOT}/gems/#{gem_name}-#{version}"
-      FileUtils.mkdir_p "#{TMP_GEM_BUILD}/#{gem_name}/lib"
-
-      FileUtils.cd "#{TMP_GEM_BUILD}/#{gem_name}" do
-        gemspec = "#{gem_name}.gemspec"
-        lib_file = "lib/#{gem_name}.rb"
-
-        File.open gemspec, "w" do |file|
-          file.puts <<-GEMSPEC
-            Gem::Specification.new do |s|
-              s.name    = #{gem_name.inspect}
-              s.version = #{version.inspect}
-              s.authors = 'Mr. Smith'
-              s.summary = 'summary'
-              s.files   = #{lib_file.inspect}
-              s.license = 'MIT'
-              s.homepage = 'http://github.com/thoughtbot/#{gem_name}'
-              s.required_ruby_version = '>= 2.3.0'
-            end
-          GEMSPEC
-        end
-
-        File.open lib_file, "w" do |file|
-          file.puts "$#{gem_name}_version = '#{version}'"
-        end
-
-        redirect = ENV["VERBOSE"] ? "" : "2>&1"
+    if opts.respond_to?(:has_key?)
+      version = opts[:version] if opts.has_key?(:version)
+      opts[:skip_build] = false unless opts.has_key?(:skip_build)
+      opts[:skip_install] = opts[:skip_build] unless opts.has_key?(:skip_install)
+    else
+      version = case opts
+      when String then opts
+      when nil then "1.0.0"
+      else
+        raise ArgumentError, "Unexpected value for opts (must be version string or options Hash): #{opts}"
+      end
+      opts = DEFAULT_OPTS.dup
+    end
+    skip_build = opts[:skip_build]
+    skip_install = opts[:skip_install]
+    return if File.exist? "#{TMP_GEM_ROOT}/gems/#{gem_name}-#{version}"
+
+    build_dir = "#{TMP_GEM_BUILD}/#{gem_name}"
+    FileUtils.mkdir_p "#{build_dir}/lib"
+
+    FileUtils.cd build_dir do
+      gemspec = "#{gem_name}.gemspec"
+      lib_file = "lib/#{gem_name}.rb"
+
+      File.open gemspec, "w" do |file|
+        file.puts <<-GEMSPEC.strip_heredoc.rstrip
+          Gem::Specification.new do |s|
+            s.name    = #{gem_name.inspect}
+            s.version = #{version.inspect}
+            s.authors = 'Mr. Smith'
+            s.summary = 'summary'
+            s.files   = #{lib_file.inspect}
+            s.license = 'MIT'
+            s.homepage = 'http://github.com/thoughtbot/#{gem_name}'
+            s.required_ruby_version = '>= 1.8.7'
+          end
+        GEMSPEC
+      end
+
+      File.open lib_file, "w" do |file|
+        file.puts "$#{gem_name}_version = '#{version}'"
+      end
+
+      redirect = ENV["VERBOSE"] ? "" : "2>&1"
 
+      # Caller may turn it into a git repo here
+      yield build_dir, redirect if block_given?
+
+      unless skip_build
```

Comment:
The git sourced gems do not need to be built or installed, so skipping those steps makes the build run a little faster.